app.rb

```diff
@@ -54,7 +52,21 @@ def initialize
       erb :"basket/show"
     end
 
-    post "/basket" do
+    post "/basket/update" do
+      begin
+        UpdateBasketItem.new(params).call
+      rescue QuantityLevelError
+        flash[:error] = "Not enough amount of product is available."
+      rescue NegativeQuantityError
+        flash[:error] = "Invalid quantity value."
+      rescue ProductNotFound
+        halt 404
+      end
+
+      redirect back
+    end
+
+    post "/basket/add" do
```

Comment:
POST method is enough you don't need to add anything to URL, so you can have

GET /basket
POST /basket
PUT /basket
DELETE /basket

no need to add anything more to url